### PR TITLE
Rag tweaks

### PR DESCRIPTION
**Pull request by SierraKomodo**
Port of https://github.com/Baystation12/Baystation12/pull/29880

Also added a global `has_danger_grab()` proc to mobs that mimics the check handcuffs used.

### CODE REVIEW

File: code/modules/mob/mob_helpers.dm [hunk 1]
```diff
@@ -35,6 +35,19 @@
 /mob/living/carbon/human/isMonkey()
 	return istype(species, /datum/species/monkey)
 
+
+/**
+ * Checks if the target has a grab from the user
+ */
+/mob/proc/has_grab(mob/user)
```
Comment (MistakeNot4892) [inline code/modules/mob/mob_helpers.dm]: `has_upgraded_grab` would be more indicative I think.

Comment (Andrew-Fall) [reply code/modules/mob/mob_helpers.dm]: has_grab(user, grab_level)?

Comment (MistakeNot4892) [reply code/modules/mob/mob_helpers.dm]: That might be a better approach yeah, some flag-based system.

Comment (SierraKomodo) [reply code/modules/mob/mob_helpers.dm]: Updated the proc name to `has_danger_grab()` since the chain specifically checks the danger var.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**